probe-rs/src/probe/stlink/usb_interface.rs

```diff
@@ -100,23 +100,17 @@ pub(crate) trait StLinkUsb: std::fmt::Debug {
 impl StLinkUsbDevice {
     /// Creates and initializes a new USB device.
     pub fn new_from_selector(selector: &DebugProbeSelector) -> Result<Self, ProbeCreationError> {
-        let context = Context::new()?;
-
-        tracing::debug!("Acquired libusb context.");
-
-        let device = context
-            .devices()?
-            .iter()
+        let device = nusb::list_devices()
+            .map_err(ProbeCreationError::Usb)?
             .filter(is_stlink_device)
             .find_map(|device| {
-                let descriptor = device.device_descriptor().ok()?;
                 // First match the VID & PID.
-                if selector.vendor_id == descriptor.vendor_id()
-                    && selector.product_id == descriptor.product_id()
+                if selector.vendor_id == device.vendor_id()
+                    && selector.product_id == device.product_id()
                 {
                     // If the VID & PID match, match the serial if one was given.
-                    if let Some(serial) = selector.serial_number.as_ref() {
-                        let sn_str = read_serial_number(&device, &descriptor).ok();
+                    if let Some(serial) = &selector.serial_number {
```

Comment:
Why do you prefer `&` when we already have `.as_ref()` two lines down?

Reply:
no reason at all, changed it back to `.as_ref()`